deliverables/common_problems.md

```diff
@@ -11,6 +11,34 @@ Boost 1.66 fails to find Python correctly when building with Python support.
 The easiest fix we (DNEG) found was to upgrade Boost to 1.70,
 as this is the version of Boost defined in CY2020.
 
+## Using Boost::Python Shared Libraries in Boost 1.70 via CMake
+
+This version of Boost is the first in the VFX Reference Platform to contain
+CMake config files that get used in calls to `find_package(Boost)`
+over the FindBoost module built into CMake.
+However this introduces a change in behaviour where linking against Boost's
+static libraries will be preferred unless
+[`BUILD_SHARED_LIBS`](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html)
+is set to `ON`.
+Setting `Boost_USE_STATIC_LIBS=OFF` is supposed to be enough to turn this
+behaviour off, however this version of Boost also contains a bug
+such that static libraries will be used unless `BUILD_SHARED_LIBS` is set to
+`ON`: https://lists.boost.org/Archives/boost/2019/06/246338.php
+
+To rectify this problem, either apply the changes mentioned in the Boost mailing
+list
+(https://github.com/boostorg/boost_install/commit/160c7cb2b2c720e74463865ef0454d4c4cd9ae7c),
+skip using the Boost config and use FindBoost instead
+(`Boost_NO_BOOST_CMAKE=ON` and `find_package(Boost MODULE ...)`),
+or set `BUILD_SHARED_LIBS=ON` in the libraries where it is appropriate.
+
+```cmake
+set(Boost_USE_STATIC_LIBS OFF)
+# Boost 1.70.0 has a bug that requires BUILD_SHARED_LIBS to be set as well:
+# https://lists.boost.org/Archives/boost/2019/06/246338.php
+set(BUILD_SHARED_LIBS ON)
+```
+
```

Comment:
These are great notes; it looks like we avoided this issue at wdas because we are using `BUILD_SHARED_LIBS=ON`. Thanks for writing these up.

Another subtle thing with boost is that the boost-mt vs. boost-non-mt settings need to consistent across usd, DCCs, and all libraries so a key setting is also:

```
    # Use Multithreaded Boost libraries (for USD/tf compatibility)
    set(Boost_USE_MULTITHREADED ON)
```

(the comment above mentions usd, but it really applies to all boost-dependent libraries)

For a while we were building boost to provide both static and shared libs. We finally got rid of all of the static libs and all of the non-mt libs as well to really make sure that we're using it correctly across all builds.

I wonder if it'd be worth mentioning a preference towards use of the shared libs in a separate PR? I can make a note to myself to follow-up with a note to suggest building boost with shared libs only if you think that'd be helpful.  The `./bootstrap.sh` args when building boost are `link=shared` and `runtime-link=shared` in order to do that. off-topic for this PR though ;-)

Reply:
That makes sense to me! Usage of static libraries are more of a "use them if you know what you're doing" type situation for us, which means that we don't have anyone using them :P